src/application/pokegear/configure/overlay_101_021EED44.c

```diff
@@ -0,0 +1,387 @@
+        PokegearConfigure_SetInitialCursorState(configureApp);
+        configureApp->substate = 0;
+        return TRUE;
+    }
+    ++configureApp->substate;
+    return FALSE;
+}
+
+BOOL PokegearConfigure_UnloadGFX(PokegearConfigureAppData *configureApp) {
+    PokegearConfigure_UnloadContextMenuText(configureApp);
+    PokegearConfigure_RemoveCursor(configureApp);
+    PokegearConfigure_DeleteSprites(configureApp);
+    PokegearConfigure_DestroySpriteManager(configureApp);
+    PokegearConfigure_UnloadGraphics(configureApp);
+    PokegearConfigure_UnloadBGs(configureApp);
+    return TRUE;
+}
+
+int PokegearConfigure_ContextMenu(PokegearConfigureAppData *configureApp) {
+    int input;
+
+    input = TouchscreenListMenu_HandleInput(configureApp->contextMenu);
+    if (input != LIST_NOTHING_CHOSEN) {
+        configureApp->pokegear->menuInputState = (MenuInputState)TouchscreenListMenu_WasLastInputTouch(configureApp->contextMenu);
+        TouchscreenListMenu_Destroy(configureApp->contextMenu);
+        PokegearConfigure_ToggleButtonFocusState(configureApp, 0, FALSE);
+        PokegearCursorManager_SetCursorSpritesAnimateFlag(configureApp->pokegear->cursorManager, 0xFFFF, TRUE);
+        if (input == 0) {
+            return PGCONF_MAIN_STATE_SWAP_SKINS;
+        } else {
+            return PGCONF_MAIN_STATE_HANDLE_INPUT;
+        }
+    }
+
+    return PGCONF_MAIN_STATE_CONTEXT_MENU;
+}
+
+static void PokegearConfigure_LoadAndSetSkin(PokegearConfigureAppData *configureApp) {
+    PokegearConfigure_UnloadGraphics_Internal(configureApp);
+    configureApp->skin = configureApp->selectedSkin;
+    configureApp->pokegear->skin = configureApp->skin;
+    PokegearApp_LoadSkinGraphics(configureApp->pokegear, configureApp->skin);
+    ov100_021E6A58(configureApp->pokegear->unk_094, configureApp->skin);
+    PokegearConfigure_LoadGraphics_Internal(configureApp);
+    PokegearConfigure_LoadPalettes(configureApp, FALSE);
+    PokegearConfigure_DrawUnlockedSkinsButtons(configureApp);
+    PokegearConfigure_SetNewSkin(configureApp, configureApp->skin);
+}
+
+BOOL PokegearConfigure_SwapSkins(PokegearConfigureAppData *configureApp) {
+    switch (configureApp->substate) {
+    case 0:
+        BeginNormalPaletteFade(1, 4, 4, RGB_BLACK, 6, 1, configureApp->heapId);
+        break;
+    case 1:
+        if (!IsPaletteFadeFinished()) {
+            return FALSE;
+        }
+        PokegearConfigure_LoadAndSetSkin(configureApp);
+        break;
+    case 2:
+        BeginNormalPaletteFade(2, 3, 3, RGB_BLACK, 6, 1, configureApp->heapId);
+        break;
+    case 3:
+        if (!IsPaletteFadeFinished()) {
+            return FALSE;
+        }
+        configureApp->substate = 0;
+        return TRUE;
+    }
+
+    ++configureApp->substate;
+    return FALSE;
+}
+
+static void PokegearConfigure_InitBGs(PokegearConfigureAppData *configureApp) {
+    int i;
+    GX_SetGraphicsMode(GX_DISPMODE_GRAPHICS, GX_BGMODE_0, GX_BG0_AS_2D);
+
+    {
```

Comment:
is there a reason for this empty scope?

Reply:
it's standard for these types of routines